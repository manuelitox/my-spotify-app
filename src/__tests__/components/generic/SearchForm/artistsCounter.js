import React from 'react'
import { shallow } from 'enzyme'

import ArtistMatchesCounter from 'components/generic/SearchForm/artistsCounter'

describe('Component | ArtistMatchesCounter', () => {
  let component 
  beforeEach( () => {
    component = shallow(<ArtistMatchesCounter isOpen={ true } loading={ false } numberOfArtists={ 4 } />)
  })
  it('should render a component', () => {
    expect(component).toMatchSnapshot()
    expect(component.text()).toEqual('4 matches')
  })
  it('should render null because isOpen is false', () => {
    component = shallow(<ArtistMatchesCounter isOpen={ false } loading={ false } numberOfArtists={ 4 } />)    
    expect(component).toMatchSnapshot()
  })
  it('should render null because loading is true', () => {
    component = shallow(<ArtistMatchesCounter isOpen={ true } loading={ true } numberOfArtists={ 4 } />)
    expect(component).toMatchSnapshot()
  })  
})