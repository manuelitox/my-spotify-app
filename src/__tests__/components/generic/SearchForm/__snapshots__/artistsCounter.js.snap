// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component | ArtistMatchesCounter should render a component 1`] = `
<styled.div>
  4
   matches
</styled.div>
`;

exports[`Component | ArtistMatchesCounter should render null because isOpen is false 1`] = `""`;

exports[`Component | ArtistMatchesCounter should render null because loading is true 1`] = `""`;
